### PR TITLE
re: Fix clippy::redundant_closure'   

### PR DESCRIPTION
**Pull request by pmnoxx**
Fix clippy::redundant_closure'   issue

### CODE REVIEW

**Comment by pmnoxx** [comment]
```
@bowenwang1996 @mm-near: /chain/ /core/ /neard/ /runtime/
@nikurt: /genesis-tools/ /test-utils/state-viewer
@matklad @posvyatokum: /test-utils/runtime-tester
@chefsale @khorolets: /chain/client/src/view_client.rs
```
@bowenwang1996 @chefsale @khorolets @matklad @mm-near @nikurt @posvyatokum Can you take a look?

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -549,7 +549,10 @@ impl ChainStore {
         &self,
         id: &CryptoHash,
     ) -> Result<Vec<ExecutionOutcomeWithIdAndProof>, Error> {
-        Ok(self.store.get_ser(ColTransactionResult, id.as_ref())?.unwrap_or_else(|| vec![]))
+        Ok(self
+            .store
+            .get_ser(ColTransactionResult, id.as_ref())?
+            .unwrap_or_else(std::vec::Vec::new))
```
Comment (matklad) [inline chain/chain/src/store.rs:555]:
```suggestion
            .unwrap_or_default())
```

How `std::vec::Vec::new` get in here? This is not something one ever writes by hand...

**Review by matklad — APPROVED** [review]
LGTM, but many cases can be further simplified to `.unwrap_or_default()`

File: chain/chain/src/store.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/chain/src/store.rs]: @matklad Oh, sorry. I fixed the issue.

Comment (pmnoxx) [reply chain/chain/src/store.rs]: @matklad I made a contribution to `clippy`, `.unwrap_or_else(std::vec::Vec::new))` should be replaced by `.unwrap_or_default())` by default

https://github.com/rust-lang/rust-clippy/pull/8163

Comment (matklad) [reply chain/chain/src/store.rs]: Awesome, thanks @pmnoxx !

**Pull request merged**